suites/e2e/tests/serial/index.js

```diff
@@ -16,33 +14,24 @@
 
 'use strict';
 
-// Test identification
-const id = `${Math.random()
-	.toString(36)
-	.substring(2, 10)}`;
+const { delay } = require('bluebird');
+const fs = require('fs').promises;
 
-module.exports = options => {
-	return {
-		id,
-		balenaOS: {
-			download: {
-				type: options.downloadType,
-				version: options.downloadVersion,
-				source: options.downloadSource,
-			},
-			network: {
-				wired: options.networkWired,
-				wireless: options.networkWireless,
-			},
-		},
-		balena: {
-			application: {
-				env: {
-					delta: options.supervisorDelta || false,
-				},
+module.exports = {
+	title: 'Serial Test',
+	tests: [
+		{
+			title: 'Recording DUT serial output',
+			run: async function (test) {
+				await this.context.get().worker.on();
+				await delay(10 * 1000);
+
+				test.not(
```

Comment:
Does this test work with the qemu worker at the moment?

Reply:
Can we run this conditionally on if the serial is connected?

Reply:
We can't detect serial connection expicitly can we?

IIRC the only way to detect a serial connection would be to actually see if the logs are coming through or not? That is also THE test. So if we have a better way to detect serial, then it would be helpful. CC: @Bucknalla 

Alternatively with #587 and the configurable bail feature, this test suite could be converted to `bail: false`. This behavior would match the current testbot-e2e-suite

Reply:
It will work with QEMU when we merge #578